### DISCUSSION TITLE
A way to configure lava flow speed using plugins.

### DISCUSSION THREAD

**Discussion by imIwKy** [PurpurMC/Purpur#1180] [__Completed__] [answered]
This would be great to reduce lag on some servers depending on specific conditions, rather than modifying it manually in the configuration file.

**Comment by YouHaveTrouble** [comment]
It would not reduce lag unless the lava spread itself was the heaviest thing that's going on. That's basically never the case.
Consider taking a spark report while the tps drops are occurring and share it on our discord, we'll help you out track down the exact cause.

**Reply by imIwKy** [reply]
Reducing lag wasn't the main reason tho. I'm trying to implement something like slowing down lava only in certain areas for example in a radius around spawn to discourage lava casting. So i was looking through the api's and didn't find anything useful in terms of events. There is FluidLevelChange event but it seems to be quite resource intensive to use. So i remembered there is a setting in purpur for slowing down lava, but i can't really use it inside a plugin.

**Reply by YouHaveTrouble** [reply]
Well, I'm commenting on what you actually typed out ;P
Changing lava speed in a world and changing lava speed in a specific region are two different pieces of cake. The current config option allows you to configure that per world, and any api added on top of that would only allow you to modify that - per world. Making it so it works for specific regions could potentially be more resource intensive than not slowing it down at all.

**Reply by imIwKy** [reply]
I see, makes sense.

**Reply by imIwKy** [reply]
Well even if only per world such a feature would be still great to have.

**Comment by BillyGalbreath** [comment, marked as answer]
You can modify Purpur's config values extremely easily by using Paper's paperweight/userdev. Then it is as simple as doing:

```java
// get the NMS level
ServerLevel level = ((CraftWorld) player.getWorld()).getHandle();
// set the lava speed in nether dimension types
level.purpurConfig.lavaSpeedNether = 5; // default's to 10
// set the lava speed in non-nether dimension types
level.purpurConfig.lavaSpeedNotNether = 10; // defaults to 30
```

We don't need an entire API for simple niche things like this, imho.
